### PR TITLE
Fix for getRuntime on Symfony older than 3.4

### PR DESCRIPTION
**Pull request by jmontoyaa**
Same issue as described here:

https://github.com/sonata-project/SonataAdminBundle/issues/4758
https://github.com/symfony/symfony/issues/24533

Similar changes:
https://github.com/sonata-project/SonataAdminBundle/pull/4762
https://github.com/sonata-project/SonataMediaBundle/pull/1325/commits/10a589f2338366e60c96945c058cfd074ee878fb

## Changelog
```markdown
### Fixed
- Fix for getRuntime on Symfony older than 3.4
```
## Subject

TwigRenderer is used as runtime until SF 3.4.
This code makes sure we retrieve the correct runtime for SF < 3.4

### CODE REVIEW

File: src/Controller/PageAdminController.php [hunk 1]
```diff
@@ -101,7 +101,11 @@ public function treeAction(Request $request = null)
         if (!method_exists('Symfony\Bridge\Twig\AppVariable', 'getToken')) {
             $twig->getExtension('Symfony\Bridge\Twig\Extension\FormExtension')->renderer->setTheme($formView, $theme);
         } else {
-            $twig->getRuntime('Symfony\Bridge\Twig\Form\TwigRenderer')->setTheme($formView, $theme);
+            // BC for Symfony < 3.4 which deprecated this runtime in favor of FormRenderer
+            if (!method_exists('Symfony\Bridge\Twig\Command\DebugCommand', 'getLoaderPaths')) {
```
Comment (greg0ire) [inline src/Controller/PageAdminController.php]: Please use the `::class` notation as much as you can.

Comment (jordisala1991) [reply src/Controller/PageAdminController.php]: It would be easier to just copy the method from AdminBundle

**Comment by jordisala1991** [comment]
You can't use directly the function because it is private, but you can copy it instead @jmontoyaa

**Comment by jmontoyaa** [comment]
@jordisala1991 Why not make it public? the same exact function is in different places:

vendor/sonata-project/media-bundle/src/Controller/GalleryAdminController.php
vendor/sonata-project/classification-bundle/src/Controller/CategoryAdminController.php
vendor/sonata-project/formatter-bundle/src/Controller/CkeditorAdminController.php
vendor/sonata-project/media-bundle/src/Controller/GalleryAdminController.php
vendor/sonata-project/media-bundle/src/Controller/MediaAdminController.php

**Comment by rvanginneken** [comment]
@jmontoyaa, @jordisala1991: How about you copy and use the function for now and discuss the cleanup of the duplicate code in a seperate issue?

**Comment by jmontoyaa** [comment]
@rvanginneken as you wish.

**Comment by greg0ire** [comment]
> Why not make it public? the same exact function is in different places:

If you do that, you now have a contract with the users that says you can't remove or change that method.

**Comment by rvanginneken** [comment]
Thanks man!

**Comment by greg0ire** [comment]
`make cs-fix` ;)

**Comment by greg0ire** [comment]
@jmontoyaa you need to run `make-cs fix`

File: src/Controller/PageAdminController.php [hunk 2]
```diff
@@ -113,7 +106,31 @@ public function treeAction(Request $request = null)
             'csrf_token' => $this->getCsrfToken('sonata.batch'),
         ]);
     }
+    
+    /**
+     * {@inheritdoc}
```
Comment (jordisala1991) [inline src/Controller/PageAdminController.php]: Please do not add this comment. I know it is maybe on other bundles, but we are now trying to remove them all

**Comment by core23** [comment]
Can you please fix the CS @jmontoyaa ?

**Comment by greg0ire** [comment]
> @jmontoyaa you need to run make-cs fix

Sorry I meant `php-cs-fixer fix`

**Comment by rvanginneken** [comment]
`/admin/sonata/page-page/tree` is broken on 3.4 over a week now. Anyway I can help this issue forward?

**Comment by greg0ire** [comment]
You can carry this PR.

File: src/Controller/PageAdminController.php (hunk 2 above)
Comment (jmontoyaa) [reply src/Controller/PageAdminController.php]: Updated.

File: src/Controller/PageAdminController.php [hunk 3]
```diff
@@ -108,7 +108,10 @@ public function treeAction(Request $request = null)
     }
     
     /**
-     * {@inheritdoc}
+     * Sets the admin form theme to form view. Used for compatibility between Symfony versions.
+     *
+     * @param FormView $formView
+     * @param string   $theme
```
Comment (greg0ire) [inline src/Controller/PageAdminController.php]: Please remove these 2 lines.

Comment (jmontoyaa) [reply src/Controller/PageAdminController.php]: 111 and 112?
113 and 114?

Comment (greg0ire) [reply src/Controller/PageAdminController.php]: The `@params` lines

**Comment by greg0ire** [comment]
Please fix the build: https://travis-ci.org/sonata-project/SonataPageBundle/jobs/319125160

File: src/Controller/PageAdminController.php [hunk 4]
```diff
@@ -17,6 +17,7 @@
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
 use Symfony\Component\Security\Core\Exception\AccessDeniedException;
+use Symfony\Component\Form\FormView;
 
 /**
  * Page Admin Controller.
```
Comment (OskarStark) [inline src/Controller/PageAdminController.php]: please remove this comment

**Comment by jordisala1991** [comment]
You have lint errors, here you can download the diff patch https://flintci.io/repositories/85/analyses/161

**Comment by jeroendk** [comment]
@jmontoyaa I tried your fix but it seems like your are missing some import statements.

```
use Symfony\Component\Form\FormView;
use Symfony\Component\Form\FormRenderer;
use Symfony\Bridge\Twig\Command\DebugCommand;
```

**Comment by gremo** [comment]
Any cache to merge this? Can I help in some way?

**Comment by core23** [comment]
Please squash your commits @jmontoyaa

**Comment by OskarStark** [comment]
Thank you @jmontoyaa

**Pull request merged**